packages/core/src/interpreter.ts

```diff
@@ -132,6 +134,10 @@ export class Interpreter<
       ...options
     };
 
+    if (resolvedOptions.state) {
+      this._initialState = this._state = resolvedOptions.state;
```

Comment:
q: why do we need both? why can't we just use `this._state`?

Reply:
Tests fail (try changing `this._initialState` in `_getInitialState()`) for `@xstate/react`. My guess is something relies on the `_initialState` (unchanged) being different from the `_state` (can change).

Reply:
I strongly suspect that those are just symptoms of something else. I prepare a PR [here](https://github.com/statelyai/xstate/pull/3874) with some changes that I think are correct. Tests fail there - but that's just because the implementation is yet incomplete. 

One of the things that React's test suite catches here:
```ts
const actor = interpreter(machine).start()
actor.stop()
actor.start()
```

If this actor has any children then they should be rehydrated when we `start` it again.